test/t8_schemes/t8_gtest_set_linear_id.cxx

```diff
@@ -0,0 +1,92 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <t8_eclass.h>
+#include <test/t8_gtest_schemes.hxx>
+#include <test/t8_gtest_custom_assertion.hxx>
+#include <test/t8_gtest_macros.hxx>
+#include "t8_gtest_dfs_base.hxx"
+
+/** In this test we iterate through all elements. 
+ * For every leaf we check if the element is equal to the element we get when setting it from the linear id.
+ * The id_counter is then increased to match the id of the next leaf.
+ */
+class class_test_set_linear_id: public TestDFS {
+  void
+  check_element () override
+  {
+    t8_debugf ("maxlvl: %i\n", maxlvl);
+    t8_debugf ("level: %d\n", scheme->element_get_level (eclass, element));
+    if (scheme->element_get_level (eclass, element) == maxlvl) {
```

Comment:
Why do we only test on maxlvl? We could also test on every level until we hit the maximum level.